comp/autoscaling/datadogclient/impl/client.go

```diff
@@ -0,0 +1,157 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+// Package datadogclientimpl implements datadog client component for querying external metrics.
+package datadogclientimpl
+
+import (
+	"sync"
+
+	datadogclient "github.com/DataDog/datadog-agent/comp/autoscaling/datadogclient/def"
+	configComponent "github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"go.uber.org/fx"
+	"gopkg.in/zorkian/go-datadog-api.v2"
+)
+
+const (
+	metricsEndpointPrefix          = "https://api."
+	metricsEndpointConfig          = "external_metrics_provider.endpoint"
+	metricsRedundantEndpointConfig = "external_metrics_provider.endpoints"
+)
+
+// endpoint represent a datadog endpoint
+type endpoint struct {
+	Site   string `mapstructure:"site" json:"site" yaml:"site"`
+	URL    string `mapstructure:"url" json:"url" yaml:"url"`
+	APIKey string `mapstructure:"api_key" json:"api_key" yaml:"api_key"`
+	APPKey string `mapstructure:"app_key" json:"app_key" yaml:"app_key"`
+}
+
+type dependencies struct {
+	fx.In
```

Comment:
Please don't use fx in the implementation `ProvideComponentConstructor` function will do the conversion for you in the `fx` folder (ie https://datadoghq.dev/datadog-agent/components/creating-components/#the-fx-folders).